pkg/machine/e2e/init_test.go

```diff
@@ -635,6 +637,94 @@ var _ = Describe("podman machine init", func() {
 		Expect(err).ToNot(HaveOccurred())
 		Expect(proc2Session.outputToString()).To(ContainSubstring("/proc/sys/fs/binfmt_misc/qemu-x86_64"))
 	})
+
+	It("init machine with invalid --provider for platform", func() {
+		var providerOverride string
+		switch testProvider.VMType() {
+		case define.QemuVirt, define.AppleHvVirt, define.LibKrun:
+			providerOverride = "wsl"
+		case define.WSLVirt, define.HyperVVirt:
+			providerOverride = "applehv"
+		default:
+			providerOverride = "unknown"
+		}
+		Expect(providerOverride).ToNot(Equal("unknown"))
```

Comment:
Might be easier to just use Fail() in the default case
`Fail("unsupported provider in tests")`